client/kubernetes/src/main/java/org/jolokia/kubernetes/client/KubernetesJmxConnector.java

```diff
@@ -0,0 +1,229 @@
+package org.jolokia.kubernetes.client;
+
+import com.squareup.okhttp.Credentials;
+import com.squareup.okhttp.Response;
+import io.kubernetes.client.ApiClient;
+import io.kubernetes.client.ApiException;
+import io.kubernetes.client.Configuration;
+import io.kubernetes.client.Pair;
+import io.kubernetes.client.apis.CoreV1Api;
+import io.kubernetes.client.models.V1OwnerReference;
+import io.kubernetes.client.models.V1Pod;
+import io.kubernetes.client.models.V1Service;
+import io.kubernetes.client.util.ClientBuilder;
+import io.kubernetes.client.util.KubeConfig;
+import java.io.FileReader;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.management.remote.JMXConnector;
+import javax.management.remote.JMXServiceURL;
+import org.jolokia.client.J4pClient;
+import org.jolokia.client.J4pClientBuilderFactory;
+import org.jolokia.client.jmxadapter.JolokiaJmxConnector;
+import org.jolokia.client.jmxadapter.RemoteJmxAdapter;
+
+public class KubernetesJmxConnector extends JolokiaJmxConnector {
+
+  private static Pattern POD_PATTERN = Pattern
+      .compile("/api/v1/namespaces/([^/]+)/pods/([^/]+)/proxy/(.+)");
```

Comment:
Maybe instead of exposing the K8A API path literally a more concise and shorter JMX connector URL would make sense, something like `service:jmx:kubernetes:pod/mynamespace/mypodname:port/path` and `service:jmx:kubernetes:service/mynamespace/mysvname:port/path` and then translate that to the full K8s API path ?

This might be more convenient for the user on expense to be less flexible (but I'm pretty sure that the API path format won't change any time soon for Kubernetes)

Reply:
Nice input. Will think about it.

Reply:
Have done some investigation. Can drop the /api/v1 and proxy from the path to reduce the boilerplate. Apart from that I have not come up with any more intuitive path. The result would be `service:jmx:kubernetes:/namespaces/myns/pods/mypod/path`

Reply:
Still, wondering if we should also drop "namespaces" and "pods" , it becomes less self documenting but briefer?

Reply:
Yeah, if we skip the connection to the K8s API syntax, then I don't see a reason why not make it even shorter (as you can't reverse the path parts anyway, that style of 'named' arguments doesn't make much sense imo)

Reply:
So `service:jmx:kubernetes:///mynamespace/mypod/path` then? Sure,  can change.